File: 0060-permutation-sequence/0060-permutation-sequence.cpp
class Solution {
public:
    string getPermutation(int n, int k) {
        vector<int> numbers;
        for(int i=1;i <= n;i++){
            numbers.push_back(i);
        }
        k--;
        string result="";
        int fact=1;
        for(int i=1;i<n;i++){
            fact *= i;
        }
        while(!numbers.empty()){
            int index=k/fact;
            result += to_string(numbers[index]);
            numbers.erase(numbers.begin()+index);
            if(numbers.empty())break;
            k %= fact;
            fact /=numbers.size();
            
        }
        return result;
    }
};

File: 0060-permutation-sequence/0060-permutation-sequence.py
class Solution(object):
    def getPermutation(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: str
        """
        k -= 1
        nums=[i for i in range(1,n+1)]
        factorials=[1]*n
        for i in range(1,n):
            factorials[i]=factorials[i-1]*i
        result=[]
        for i in range(n-1,-1,-1):
            idx=k//factorials[i]
            result.append(str(nums[idx]))
            nums.pop(idx)
            k %= factorials[i]
        return "".join(result)
        